plutus-benchmark/script-contexts/test/Spec.hs

```diff
@@ -41,7 +41,7 @@ testCheckSc1 = testGroup "checkScriptContext1"
         compiledCodeToTerm $ mkCheckScriptContext1Code (mkScriptContext 4)
     , testCase "fails on 5" . assertFailed $
         compiledCodeToTerm $ mkCheckScriptContext1Code (mkScriptContext 5)
-    , Tx.fitsInto "checkScriptContext1 (size)" (mkCheckScriptContext1Code (mkScriptContext 1)) 3721
+    , Tx.fitsInto "checkScriptContext1 (size)" (mkCheckScriptContext1Code (mkScriptContext 1)) 3851
```

Comment:
some regression here.

Reply:
Any idea why? I can't think of any reason why this PR makes anything bigger or more expensive.

Reply:
I suspect `costIsOk` isn't quite right. I can't think of anything else.

Reply:
I will do some more experiments changing what `costIsOk` checks.

Reply:
You can also look at the readable PIR before and after (use `goldenPirReadable` to get them). The readable PIRs are fairly readable.

Reply:
Maybe I'm doing something wrong but the golden file from the master branch is the same as the one from this branch.

Reply:
You may have forgot to `-fforce-recomp`.

Reply:
I just remembered another possible reason for the regression: `safeToBetaReduce` is fixed in this PR also. The last PR that fixed it did also cause some regression.

Reply:
I'll add some tests to `termSize`. It seems we don't have any, just a couple of doc tests for `kindSize`. These seems to all involve type applications too so I'll make sure to check that.

Reply:
> It is inlining some things with bigger size

That "some things" is `t`, which should not be inlined because it increases the size. So it seems the inliner isn't checking the sizes properly.

Reply:
While adding tests is good, I would try to figure out what exactly happened in this case, and then add some tests accordingly (if this is what you are doing, great), rather than just adding some tests and hoping some of them reveal the issue.